### PR TITLE
Add full Base58Check validation (Standard + Cold)

### PR DESCRIPTION
**Pull request by JSKitty**
## Abstract

This PR finally adds [Base58Check](https://en.bitcoin.it/wiki/Base58Check_encoding#Features_of_Base58Check), instead of our previous "soft checks" which were simple Prefix and Length checks, now, all address payloads are checksummed and validated properly, so no longer will MPW accept addresses after changing a letter (since the checksum would invalidate).

To validate Cold Addresses, I've imported PIVX Core's `STAKING_ADDRESS` byte for [Mainnet](https://github.com/PIVX-Project/PIVX/blob/f02ab524131e79805910c846a8569ba3beea9145/src/chainparams.cpp#L330C12-L330C12) and [Testnet](https://github.com/PIVX-Project/PIVX/blob/f02ab524131e79805910c846a8569ba3beea9145/src/chainparams.cpp#L474C26-L474C26).

No more blind validation! - Jest tests were also added.

---

## Testing
To test this PR, it's suggested to attempt these user flows, or variations of these:
- Get a valid address, change it's last letter (or a random letter), ensure MPW's Send menu rejects it.
- Get a valid Cold Staking address, change it's letter(s) as above, try setting it as your Cold Staking Address.
- Try setting a Contact with an invalid address, it should reject.

If any errors are found, the PR works unexpectedly, or you have viable suggestions to improve the UX or functionality of the PR, let me know!

---

### CODE REVIEW

File: tests/unit/masterkey.spec.js [hunk 1]
```diff
@@ -140,6 +140,26 @@ describe('mainnet tests', () => {
             }
         }
     });
+    test('Correct Base58Check validation of addresses', async () => {
```
Comment (Duddino) [inline tests/unit/masterkey.spec.js:143]:
```suggestion
    test('Correct Base58Check validation of addresses', () => {
```

File: tests/unit/masterkey.spec.js [hunk 2]
```diff
@@ -227,4 +247,24 @@ describe('testnet tests', () => {
...
```
Comment (Duddino) [inline tests/unit/masterkey.spec.js:250]:
```suggestion
    test('Correct Base58Check validation of addresses', () => {
```

**Review by Duddino — APPROVED** [review]
tACK with a really minor nitpick

**Review by panleone — APPROVED** [review]
tACK bfd7a475a69dae3da077864c88320130d701633f up to failing tests

**Pull request merged**